src/hidra/receiver/asapo_transfer.py

```diff
@@ -161,12 +169,17 @@ def construct_config(config_path, identifier):
     return TransferConfig(**config)
 
 
-def run_transfer(asapo_transfer, timeout=3):
+def run_transfer(asapo_transfer, config, timeout=3):
     while True:
         try:
             asapo_transfer.run()
         except Stopped:
             break
+        except zmq.error.ZMQError:
+            logger.warning("Running Transfer stopped with an exception", exc_info=True)
+            logger.info("Retrying connection with different port")
+            asapo_transfer.query = create_query(config.signal_host, config.detector_id)
```

Comment:
Better do not mutate the existing `AsapoTransfer` object but create a new one (you could, e.g.,  create a closure `create_asapo_transfer` with the port number as the only argument).

Reply:
`AsapoTransfer` is recreated now.